transaction/core/src/account_keys.rs

```diff
@@ -360,10 +363,32 @@ impl AccountKey {
             fog_authority_sig: Default::default(),
         };
 
-        // Compute fog_authority_sig as a signature over self.fog_authority_key_fingerprint
+        // NOTE: The fog_authority_sig is deterministic due to using the private key hash as the rng seed
         if !self.fog_report_url.is_empty() {
-            // FIXME: FOG-106 fog_authority_sig should be a Schnorrkel sig using subaddress_view_private
-            result.fog_authority_sig.extend(&[9u8, 9u8, 9u8, 9u8]);
+            // Construct the fog authority signature over the fingerprint using the view privkey
+            let view_private = self.subaddress_view_private(index);
+            let scalar: &Scalar = view_private.as_ref();
+
+            // Nonce is hash( private_key || message )
+            let nonce_digest: Vec<u8> = [
+                &scalar.to_bytes(),
+                self.fog_authority_key_fingerprint.as_slice(),
+            ]
+            .concat();
+            let nonce: [u8; 32] = Sha3_256::digest(&nonce_digest).into();
```

Comment:
I have a slight preference for using Blake2B here instead of adding a new dependency on sha3 ("mo dependencies mo problems", as Biggie would say), but this is fine.